src/commands/moderate.ts

```diff
@@ -0,0 +1,305 @@
+import { MessageEmbed, TextChannel } from 'discord.js';
+import { logger } from '../utilities/logger';
+import { CommandDef } from './command-def';
+
+export const moderate: CommandDef = {
+  prefix: 'moderate',
+  description:
+    'This handles the moderation commands. It can create a `private` discussion channel, `suspend` a user, and `close` a private/suspended channel.',
+  usage:
+    'moderate <suspend|private> <@user> [reason] || moderate close <accepted|denied> <@user>',
+  command: async (message, { config }) => {
+    // Extract values
+    const { author, member, content, channel, mentions, guild } = message;
+    const {
+      LOG_MSG_CHANNEL,
+      SUSPEND_ROLE,
+      MOD_ROLE,
+      SUSPEND_CATEGORY,
+      BOT_ROLE
+    } = config;
+
+    // Error handling
+    if (!guild) {
+      logger.error('Guild error.');
+      return;
+    }
```

Comment:
How when is this possible? 🤔

Reply:
DM messages have no `guild` property, I believe.

This is an error handling trick I learned with my own bot, to avoid optional chaining everywhere that `guild` is accessed.